hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonOutputCommitter.java

```diff
@@ -234,13 +234,7 @@ private void commitJobFinal(JobContext context, CarbonLoadModel loadModel,
     String segmentsToBeDeleted =
         context.getConfiguration().get(CarbonTableOutputFormat.SEGMENTS_TO_BE_DELETED, "");
     List<Segment> segmentDeleteList = Segment.toSegmentList(segmentsToBeDeleted.split(","), null);
-    Set<Segment> segmentSet = new HashSet<>();
-    if (updateTime != null || uniqueId != null) {
-      segmentSet = new HashSet<>(
-          new SegmentStatusManager(carbonTable.getAbsoluteTableIdentifier(),
-              context.getConfiguration()).getValidAndInvalidSegments(carbonTable.isMV())
-                  .getValidSegments());
-    }
+    Set<Segment> segmentSet = Collections.singleton(loadModel.getSegment());
```

Comment:
segmentSet can be moved inside `  if (updateTime != null || uniqueId != null) {` check. If updateTime and uniqueId is null, this value will be unused

Reply:
@Indhumathi27  Thanks for reviewing, I will improve this.
In fact, I have not fully grasped the scope of influence, so I'm working on this and improving the unit tests.